### PR TITLE
Allow vaults to have values stick to registry defaults

### PR DESCRIPTION
**Pull request by ben-kaufman**

### CODE REVIEW

File: contracts/vaults/Base.sol [hunk 1]
```diff
@@ -222,6 +253,54 @@ contract Base is ERC4626Upgradeable, OwnableUpgradeable, ReentrancyGuardUpgradea
...
+            return arbitrationConfig.arbitrator;
+        } else {
+            return registry.defaultArbitrator();
+        }
+    }
+
+    /** 
+    * @notice Returns the vault challenge period
+    */
+    function getChallengePeriod() public view returns(uint256) {
```
Comment (jellegerbrandy) [inline contracts/vaults/Base.sol:285]: is it not nicer to do something like 
```
function getChallengePeriod() { return arbitrationconfig.challengePeriod || registry.getChallengePeriod())
```

Comment (ben-kaufman) [reply contracts/vaults/Base.sol:285]: How would that work?

File: contracts/HATVaultsRegistry.sol [hunk 2]
```diff
@@ -121,7 +129,11 @@ contract HATVaultsRegistry is Ownable {
     // asset => amount
     mapping(address => uint256) public governanceHatReward;
 
-    HATBountySplit public hatBountySplit;
+    HATBountySplit public defaultHATBountySplit;
+
+    address public defaultArbitrator;
```
Comment (jellegerbrandy) [inline contracts/HATVaultsRegistry.sol:134]: afaic these vars can also be just alled `arbitrator`, `challengePeriod`, etc.

Comment (ben-kaufman) [reply contracts/HATVaultsRegistry.sol:134]: You don't think it's clearer/ more accurate like this?

File: contracts/HATVault.sol [hunk 3]
```diff
@@ -91,16 +91,21 @@ contract HATVault is Claim, Deposits, Params, Withdrawals {
...
         _transferOwnership(_params.owner);
         tokenLockFactory = _registry.tokenLockFactory();
-        arbitrator = _registry.owner();
-        challengePeriod = 3 days;
-        challengeTimeOutPeriod = 5 weeks;
+
+        // Set vault to use default registry values where applicable
+        hatBountySplit = HATVaultsRegistry.HATBountySplit({
+            governanceHat: NULL_UINT,
+            hackerHatVested: 0
```
Comment (jellegerbrandy) [inline contracts/HATVault.sol:104]: no NULL_UINT here as well?

File: contracts/HATVaultsRegistry.sol (hunk 2 above)
Comment (jellegerbrandy) [reply contracts/HATVaultsRegistry.sol:134]: it can also be more confusing in some cases. As in "so this is the defaultArbitrator, where is the actual arbitrator defined?"

Comment (jellegerbrandy) [reply contracts/HATVaultsRegistry.sol:134]: it is more like a fallback value

File: contracts/HATVault.sol (hunk 3 above)
Comment (ben-kaufman) [reply contracts/HATVault.sol:104]: Well it just saves gas but gets the same result. Since it's a struct I only check the governanceHat, so making this 0 is cheaper

File: contracts/HATVaultsRegistry.sol (hunk 2 above)
Comment (jellegerbrandy) [reply contracts/HATVaultsRegistry.sol:134]: "default" is a bit ambiguous, which is why we had such different interpretations of how this should be implemented

Comment (ben-kaufman) [reply contracts/HATVaultsRegistry.sol:134]: I disagree but not very important tbh. So how you call it? Also rename the event and function name?

File: contracts/vaults/Base.sol [hunk 4]
```diff
@@ -160,14 +152,18 @@ contract Base is ERC4626Upgradeable, OwnableUpgradeable, ReentrancyGuardUpgradea
...
     bool public depositPause;
 
+    mapping(address => uint256) public withdrawEnableStartTime;
+
+    // the HATBountySplit of the vault
+    HATVaultsRegistry.HATBountySplit internal hatBountySplit;
+
     // address of the arbitrator - which can dispute claims and override the committee's decisions
-    address public arbitrator;
+    address internal arbitrator;
```
[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply contracts/vaults/Base.sol:161]: ah ok nvm :)

File: docs/roles.md [hunk 5]
```diff
@@ -5,7 +5,22 @@ The contracts in this repository define a number of different roles that have th
 
 ## `HATVaultsRegistry.owner`
 
-- `swapAndSend` and swap and send the vauit's token that are earmarked for payout (after approveClaim)
+- can call `setFeeSetter` set the feeSetter role
```
Comment (jellegerbrandy) [inline docs/roles.md:8]: Can you undo these changes?

Comment (jellegerbrandy) [reply docs/roles.md:8]: this is a merge going wrong, we can update later

File: contracts/HATVaultsRegistry.sol (hunk 2 above)
Comment (jellegerbrandy) [reply contracts/HATVaultsRegistry.sol:134]: let's just leave this for now and merge as is

**Review by jellegerbrandy — APPROVED** [review]

File: docs/roles.md (hunk 5 above)
Comment (ben-kaufman) [reply docs/roles.md]: Ok yeah something went wrong with the conflicts, I just made it as it is now
